helix-core/src/movement.rs

```diff
@@ -555,6 +558,14 @@ fn reached_target(target: WordMotionTarget, prev_ch: char, next_ch: char) -> boo
             is_sub_word_boundary(prev_ch, next_ch, Direction::Backward)
                 && (!(prev_ch.is_whitespace() || prev_ch == '_') || char_is_line_ending(next_ch))
         }
+        WordMotionTarget::EvilNextWordStart => {
+            is_word_boundary(prev_ch, next_ch)
+                && (!char_is_line_ending(next_ch) && !next_ch.is_whitespace())
```

Comment:
Hello @usagi-flow, I introduced this bit of evil specific code here, which might make rebasing/merging with upstream harder in the future.

Perhaps it's possible to get rid of it, at the expense of complicating the code in `commands/evil.rs` a bit more, though I haven't tried. Let me know what you prefer.